tubular/imputers.py

```diff
@@ -190,17 +190,22 @@ def fit(self, X: pd.DataFrame, y: pd.Series | None = None) -> pd.DataFrame:
                 # filter out null rows so their weight doesn't influence calc
                 filtered = X[X[c].notna()]
 
-                # first sort df by column to be imputed (order of weight column shouldn't matter for median)
-                filtered = filtered.sort_values(c)
+                # below algorithm only works for >1 non null values
+                if len(filtered) > 0:
```

Comment:
Think this might read more clearly if the logic is flipped i.e. 
`if len(filtered) <=0:`
`   median = np.nan`
`else:`
`...`

Then the handling of this issue is in one place rather than sandwiched around the rest of the logic.

Reply:
Sure, updated here https://github.com/lvgig/tubular/pull/202/commits